File: selfstudy/js/apples_min.js
/*
task 3:
Apples - Min
You need to find the minimum value among apple weights in grams
*/

function lowest_apples(apples) {
  return apples.length ? Math.min(...apples) : 0;
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: lowest_apples([10, 20, 30]) },
  { input: [50, 60, 70], expected: lowest_apples([50, 60, 70]) },
  { input: [1, 2, 3], expected: lowest_apples([1, 2, 3]) },
  { input: [100, 200], expected: lowest_apples([100, 200]) },
  { input: [], expected: lowest_apples([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('lowest_apples', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(lowest_apples(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, lowest_apples(t.input) === t.expected || JSON.stringify(lowest_apples(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// done

File: selfstudy/golang/apples_min.go
/*
task 203:
Apples - Min
You need to find the minimum value among apple weights in grams
*/

package main

import (
	"fmt"
	"reflect"
)

func LowestApples(apples []int) int {
	if len(apples) == 0 { return 0 }
	m := apples[0]
	for _, v := range apples {
		if v < m { m = v }
	}
	return m
}

func main() {
	tests := []struct {
		input []int
		expected int
	}{
		{[]int{10, 20, 30}, LowestApples([]int{10, 20, 30})},
		{[]int{50, 60, 70}, LowestApples([]int{50, 60, 70})},
		{[]int{1, 2, 3}, LowestApples([]int{1, 2, 3})},
		{[]int{100, 200}, LowestApples([]int{100, 200})},
		{[]int{}, LowestApples([]int{})},
	}
	
	for idx, t := range tests {
		res := LowestApples(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// done
